service/gcs/errors/errors.go

```diff
@@ -45,3 +62,111 @@ func (e *processDoesNotExistError) Error() string {
 func NewProcessDoesNotExistError(pid int) *processDoesNotExistError {
 	return &processDoesNotExistError{Pid: pid}
 }
+
+// StackTracer is an interface originating (but not exported) from the
+// github.com/pkg/errors package. It defines something which can return a stack
+// trace.
+type StackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+type baseHresultError struct {
+	hresult Hresult
+}
+
+func (e *baseHresultError) Error() string {
+	return fmt.Sprintf("HRESULT: %d", e.Hresult())
+}
+func (e *baseHresultError) Hresult() Hresult {
+	return e.hresult
+}
+func (e *baseHresultError) Format(s fmt.State, verb rune) {
```

Comment:
I think you just need to implement Error(). There's no need to be more explicit since Go guarantees certain behaviors.

https://golang.org/pkg/fmt/ explains the algorithm:

> Except when printed using the verbs %T and %p, special formatting considerations apply for operands that implement certain interfaces. In order of application:
> 
> 1. If the operand is a reflect.Value, the operand is replaced by the concrete value that it holds, and printing continues with the next rule.
> 
> 2. If an operand implements the Formatter interface, it will be invoked. Formatter provides fine control of formatting.
> 
> 3. If the %v verb is used with the # flag (%#v) and the operand implements the GoStringer interface, that will be invoked.
> 
> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
> 
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
> 
> 5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
>